File: appverse-backend/src/main/java/com/appverse/service/ReviewService.java
package com.appverse.service;

import com.appverse.dto.ReviewDTO;

import java.util.List;

public interface ReviewService {
    ReviewDTO createReview(ReviewDTO reviewDTO, Long userId);
    ReviewDTO updateReview(Long id, ReviewDTO reviewDTO, Long userId);
    void deleteReview(Long id, Long userId);
    List<ReviewDTO> getReviewsByApp(Long appId);
    List<ReviewDTO> getReviewsByUser(Long userId);
    ReviewDTO getReviewById(Long id);
}

File: appverse-frontend/src/services/reviewService.js
import api from './api';

const reviewService = {
  getReviewsByApp: async (appId) => {
    const response = await api.get(`/reviews/app/${appId}`);
    return response.data;
  },

  getMyReviews: async () => {
    const response = await api.get('/reviews/user');
    return response.data;
  },

  createReview: async (reviewData) => {
    const response = await api.post('/reviews', reviewData);
    return response.data;
  },

  updateReview: async (id, reviewData) => {
    const response = await api.put(`/reviews/${id}`, reviewData);
    return response.data;
  },

  deleteReview: async (id) => {
    const response = await api.delete(`/reviews/${id}`);
    return response.data;
  },
};

export default reviewService;
